### ISSUE TITLE
API calls that could receive an "empty" object will incorrectly appear as an error`

### ISSUE THREAD

**Issue by heartsucker** [freedomofpress/securedrop-client#71]
>  Note: I may be understanding this pattern poorly, since so far I have been wrong with basically everything about the client.

Consider this snippet in the `APICallRunner`.

```python
    def call_api(self):
        """
        Call the API. Emit a boolean signal to indicate the outcome of the
        call. Timeout signal emitted after 5 seconds. Any return value or
        exception raised is stored in self.result.
        """
        self.timer = QTimer()
        self.timer.timeout.connect(lambda: self.timeout.emit())
        self.timer.setSingleShot(True)
        self.timer.start(5000)
        try:
            logger.info('Calling API with "{}" method'.format(
                        self.api_call.__name__))
            self.result = self.api_call(*self.args, **self.kwargs)
            result_flag = bool(self.result)
        except Exception as ex: 
            logger.error(ex)
            self.result = ex
            result_flag = False
        self.call_finished.emit(result_flag)
```

If we pass that the SDK function `Client.get_replies_from_source`, the response from that is of type `List[Reply]`. if the list is empty, the `call_api` function will bool-ify the result to `False`, but this is not actually an error.

**Comment by ntoll** [comment]
OK... so I'll make a fix for this stacked on #77 as per my suggestions in #80.

**Comment by redshiftzero** [comment]
closed by  #85
